File: problems/easy/Arrays/26.Remove Duplicates from Sorted Array/remove_duplicates_from_sorted_array.py
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # O(n) time | O(1) space
        k = 0
        for i in range(len(nums)):
            if nums[i] != nums[k]:
                nums[i], nums[k + 1] = nums[k + 1], nums[i]
                k += 1
        return k + 1

File: problems/easy/Arrays/26.Remove Duplicates from Sorted Array/remove_duplicates_from_sorted_array.cpp
#include <vector>

class Solution {
public:
    // O(n) time | O(1) space
    int removeDuplicates(std::vector<int>& nums) {
        int k = 0;
        for(int i = 0; i < nums.size(); ++i) {
            if(nums[i] != nums[k]) {
                swap(nums, i, k+1);
                k += 1;
            }
        }
        return k + 1;
    }

    void swap(std::vector<int>& nums, int i, int j) {
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
};